prov/sockets/src/sock_eq.c

```diff
@@ -83,13 +83,15 @@ static ssize_t sock_eq_sread(struct fid_eq *eq, uint32_t *event, void *buf,
 			SOCK_LOG_DBG("Nothing to read from eq!\n");
 			return -FI_EAGAIN;
 		}
-		ret = dlistfd_wait_avail(&sock_eq->list, timeout);
-		if (!dlistfd_empty(&sock_eq->err_list))
-			return -FI_EAVAIL;
-
-		if (ret <= 0)
-			return (ret == 0 || ret == -FI_ETIMEDOUT) ?
-				-FI_EAGAIN : ret;
+		do {
+			ret = dlistfd_wait_avail(&sock_eq->list, timeout);
+			if (!dlistfd_empty(&sock_eq->err_list))
+				return -FI_EAVAIL;
+
+			if (ret < 0)
+				return (ret == -FI_ETIMEDOUT) ?
+					-FI_EAGAIN : ret;
+		} while (ret == 0);
```

Comment:
Discussed with @jithinjosepkl. The loop was added to avoid <code>-FI_EAGAIN</code> when fd was not written to. But this is a wrong behavior of <code>fi_eq_sread</code> which was caused by a bug in <code>fi_eq_readerr</code>. It didn't reset the signal and poll returned 1 when eq was empty. Added code to reset signal after data was read from err queue. Will update the PR with the fix. Thanks Jithin for pointing this out!